src/plugins/auto_batch/src/plugin.cpp

```diff
@@ -7,353 +7,395 @@
-        IE_SET_METRIC_RETURN(SUPPORTED_METRICS, metrics);
-    } else if (name == METRIC_KEY(FULL_DEVICE_NAME)) {
-        IE_SET_METRIC_RETURN(FULL_DEVICE_NAME, _pluginName);
-    } else if (name == METRIC_KEY(SUPPORTED_CONFIG_KEYS)) {
-        IE_SET_METRIC_RETURN(SUPPORTED_CONFIG_KEYS, supported_configKeys);
-    } else {
-        IE_THROW(NotFound) << "Unsupported metric key " << name;
-    }
+std::shared_ptr<ov::ICompiledModel> Plugin::compile_model(const std::shared_ptr<const ov::Model>& model,
+                                                          const ov::AnyMap& properties) const {
+    return compile_model(model, properties, {});
 }
 
-InferenceEngine::IExecutableNetworkInternal::Ptr Plugin::LoadExeNetworkImpl(
-    const InferenceEngine::CNNNetwork& network,
-    const std::map<std::string, std::string>& user_config) {
-    return LoadNetworkImpl(network, nullptr, user_config);
-}
-
-InferenceEngine::IExecutableNetworkInternal::Ptr Plugin::LoadNetworkImpl(
-    const InferenceEngine::CNNNetwork& network,
-    const std::shared_ptr<InferenceEngine::RemoteContext> ctx,
-    const std::map<std::string, std::string>& user_config) {
-    auto core = GetCore();
+std::shared_ptr<ov::ICompiledModel> Plugin::compile_model(const std::shared_ptr<const ov::Model>& model,
+                                                          const ov::AnyMap& properties,
+                                                          const ov::RemoteContext& context) const {
+    auto core = get_core();
     if (core == nullptr) {
-        IE_THROW() << "Please, work with Auto-Batching device via InferencEngine::Core object";
+        OPENVINO_THROW("Please, work with Auto-Batching device via InferencEngine::Core object");
     }
-    auto fullConfig = mergeConfigs(_config, user_config);
-    auto device_batch = fullConfig.find(CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG));
-    if (device_batch == fullConfig.end())
-        device_batch = fullConfig.find(ov::device::priorities.name());
-    if (device_batch == fullConfig.end()) {
-        IE_THROW() << "KEY_AUTO_BATCH key is not set for BATCH device";
+
+    // merge configs from func properties and m_plugin_config
+    auto full_properties = merge_properties(m_plugin_config, properties);
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    auto device_batch = full_properties.find(CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG));
+    if (device_batch == full_properties.end())
+        device_batch = full_properties.find(ov::device::priorities.name());
+    if (device_batch == full_properties.end()) {
+        OPENVINO_THROW("ov::device::priorities key for AUTO NATCH is not set for BATCH device");
     }
-    auto metaDevice = ParseMetaDevice(device_batch->second, user_config);
-    const auto& deviceName = metaDevice.device_name;
-    const auto& deviceConfig = metaDevice.config;
-    auto deviceConfigNoAutoBatch = deviceConfig;
+    m_plugin_config[device_batch->first] = device_batch->second;
```

Comment:
potential race condition here

Reply:
Yes, u are right. If we find another way for get_default_context(), this can be removed.